tedana/utils.py

```diff
@@ -208,7 +209,21 @@ def dice(arr1, arr2, axis=None):
         dsi = np.zeros(arr_sum.shape)
     else:
         intersection = np.logical_and(arr1, arr2)
-        dsi = (2.0 * intersection.sum(axis=axis)) / arr_sum
+        # Count number of zero-elements in the denominator and report
+        total_zeros = np.count_nonzero(arr_sum == 0)
+        if total_zeros > 0:
+            LGR.warning(
+                f"Calculating dice coefficient with {total_zeros} "
+                "zero-elements in the denominator. "
+                "Please check your component table for dice columns with 0-"
+                "values"
+            )
+
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore")
+                dsi = (2.0 * intersection.sum(axis=axis)) / arr_sum
+        else:
+            dsi = (2.0 * intersection.sum(axis=axis)) / arr_sum
```

Comment:
```suggestion

        with warnings.catch_warnings():
            warnings.simplefilter("ignore")
            dsi = (2.0 * intersection.sum(axis=axis)) / arr_sum
```

Can we have the filter target the divide-by-zero warning specifically?